pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -123,7 +123,9 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetad
     }
     //inverted index columns
     invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-
+    if (sortedColumn != null && !invertedIndexColumns.contains(sortedColumn)) {
+      invertedIndexColumns.add(sortedColumn);
```

Comment:
This only happens during realtime segment indexing.

Let me explain why we need this.
1. Since this column is a sorted column(technically is also an inverted index). So we should create inverted index on the fly as data are not sorted in realtime segment buffer.
2. When we persist a segment to disk, during the persistence, we need to sort this column, it relies on the inverted index to give the iterator for matching docId. This part will be very expansive(#cardinality * #rows) if we don't have inverted index.
3. To address your concern, this column shouldn't create inverted index after persistence, so no inverted index will be created or loaded after the persistence finished.